components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -9417,7 +9421,8 @@ public void undeployAPIProductRevisionDeployment(String apiProductId, String api
         product.setEnvironments(environmentsToRemove);
         removeFromGateway(product, tenantDomain, new HashSet<>(apiRevisionDeployments),Collections.emptySet());
         apiMgtDAO.removeAPIRevisionDeployment(apiRevisionId, apiRevisionDeployments);
-        GatewayArtifactsMgtDAO.getInstance().removePublishedGatewayLabels(apiProductId, apiRevisionId);
+        GatewayArtifactsMgtDAO.getInstance().removePublishedGatewayLabels(apiProductId, apiRevisionId,
```

Comment:
if environmentsToRemove.size >0 only call this method

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/11249/commits/c66ffa3987571605b21144c974d9d9cf14f29e4f